x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -882,6 +929,37 @@ public static Authentication newApiKeyAuthentication(AuthenticationResult<User>
         return authentication;
     }
 
+    public static Authentication newRemoteAccessAuthentication(
+        AuthenticationResult<User> remoteAccessApiKeyAuthcResult,
+        Authentication authenticationFromRemoteCluster,
+        String nodeName
+    ) {
```

Comment:
I'd personally prefer a different method signature that is not `static` and similar to the `token()` method, e.g.:
```java
public static toRemoteAccess(RemoteAccessAuthentication remoteAccessAuthentication ) {
  assert isApiKey(); 
  ...
}
```

I think it works better with composition of authenticatorChain. It will move some logic around remote authentication handling elsewhere into this method. But they feel like good changes to me.

Reply:
I like your suggestion. Clean!